src/GDBDebugSession.ts

```
@@ -1225,11 +1225,10 @@ export class GDBDebugSession extends LoggingDebugSession {
 
             this.sendResponse(response);
         } catch (err) {
-            this.sendErrorResponse(
-                response,
-                1,
-                err instanceof Error ? err.message : String(err)
-            );
+            // For multiple debug session, another sessions can catch error in case of not have variable in source.
```

No, that is not correct, if there is an error the adapter must report it as an error. The caller can decide to ignore the error and call other adapters if it wants to in a multi-session.

Note that the tests now fail with this change, in particular look at tests `should reject evaluation of expression without a frame` and `should reject evaluation of invalid expression`

If there is additional information needed in the error, please feel free to provide a PR that does that.

---

I understood. I will find a different solution for this. Thanks for your comment.